src/background/services/monetization.ts

```diff
@@ -49,16 +50,17 @@ export class MonetizationService {
       return
     }
     const { tabId, frameId } = getSender(sender)
+    const sessions = this.sessions[tabId]
 
-    if (this.sessions[tabId] == null) {
+    if (sessions == null) {
       this.sessions[tabId] = new Map()
     }
 
-    const sessionsCount = this.sessions[tabId].size + payload.length
+    const sessionsCount = sessions.size + payload.length
```

Comment:
Bug: If `sessions` is not set initially, we're not setting it to `new Map()` in the if block. Consider assigning `const sessions` after the if block.